components/pd_client/src/util.rs

```diff
@@ -611,18 +624,29 @@ impl PdConnector {
             for ep in m.get_client_urls() {
                 match self.connect(ep.as_str()).await {
                     Ok((_, r)) => {
-                        let new_cluster_id = r.get_header().get_cluster_id();
-                        if new_cluster_id == cluster_id {
-                            // check whether the response have leader info, otherwise continue to
-                            // loop the rest members
-                            if r.has_leader() {
-                                return Ok(r);
-                            }
+                        let header = r.get_header();
+                        // Try next follower endpoint if the cluster has not ready since this pr:
+                        // pd#5412.
+                        if let Err(e) = check_resp_header(header) {
+                            error!("connect pd failed";"endpoints" => ep, "error" => ?e);
                         } else {
-                            panic!(
-                                "{} no longer belongs to cluster {}, it is in {}",
-                                ep, cluster_id, new_cluster_id
-                            );
+                            let new_cluster_id = header.get_cluster_id();
+                            if new_cluster_id == cluster_id {
```

Comment:
Is it possible that the `cluster_id` from `previous` is 0 as well

Reply:
the previouse can not be zero if tikv has connect it. otherwise , tikv only connect the pd if the cluster id is not zero.

Reply:
yes, the previous cluster id may be zero , so  client  can connect the the new cluster id.